docs/concepts/node.rst

```diff
@@ -0,0 +1,94 @@
+====
+Node
+====
+
+The Node is the basic unit of Hamilton. Each node contains code to execute, and they can be assembled to form a dataflow.
```

Comment:
```suggestion
You as a user write functions, that Hamilton ingests to create nodes in a dataflow, a.k.a. directed acyclic graph. Nodes are the unit of work that Hamilton thinks and operates in.  So it is common to for users to interchangeably use "function" and "node" as synonyms for each other, since they have a duality. However, you as a user never directly create "nodes" in Hamilton. In this section we'll describe how functions become nodes, that then form a dataflow. 
```